modules/gapi/misc/python/samples/gaze_estimation.py

```diff
@@ -0,0 +1,403 @@
+    parser.add_argument('--gazed',
+                        default='CPU',
+                        help='Target device for the gaze vector estimation inference ' +
+                        '(e.g. CPU, GPU, VPU, ...)')
+    parser.add_argument('--eyem',
+                        default='open-closed-eye-0001.xml',
+                        help='Path to OpenVINO open closed eye model (.xml)')
+    parser.add_argument('--eyed',
+                        default='CPU',
+                        help='Target device for the eyes state inference (e.g. CPU, GPU, VPU, ...)')
+    return parser
+
+
+# ------------------------Kernels------------------------
+@cv.gapi.op('custom.ProcessPoses', in_types=[cv.GArray.GMat,
+                                             cv.GArray.GMat,
+                                             cv.GArray.GMat],
+            out_types=[cv.GArray.GMat])
+class ProcessPoses:
+    @staticmethod
+    def outMeta(arr_desc0, arr_desc1, arr_desc2):
+        return cv.empty_array_desc()
+
+
+@cv.gapi.kernel(ProcessPoses)
+class GProcessPosesImpl:
+    @staticmethod
+    def run(in_ys, in_ps, in_rs):
+        out_poses = []
+        size = len(in_ys)
+        for i in range(size):
+            out_poses.append(np.array([in_ys[i][0], in_ps[i][0], in_rs[i][0]]).T)
+        return out_poses
+
+
+@cv.gapi.op('custom.ParseEyes', in_types=[cv.GArray.GMat,
+                                          cv.GArray.Rect,
+                                          cv.GOpaque.Size],
+            out_types=[cv.GArray.Rect,
+                       cv.GArray.Rect,
+                       cv.GArray.Point,
+                       cv.GArray.Point])
+class ParseEyes:
+    @staticmethod
+    def outMeta(arr_desc0, arr_desc1, arr_desc2):
+        return cv.empty_array_desc(), cv.empty_array_desc(), \
+               cv.empty_array_desc(), cv.empty_array_desc()
+
+
+@cv.gapi.kernel(ParseEyes)
+class GParseEyesImpl:
+    @staticmethod
+    def run(in_landm_per_face, in_face_rcs, frame_size):
+        left_eyes = []
+        right_eyes = []
+        midpoints = []
+        lmarks = []
+        num_faces = len(in_landm_per_face)
+        surface = (0, 0, *frame_size)
+        for i in range(num_faces):
+            rect = in_face_rcs[i]
+            points = process_landmarks(*rect, in_landm_per_face[i])
+            for p in points:
+                lmarks.append(p)
+            size = int(len(in_landm_per_face[i][0]) / 2)
+
+            rect, midpoint_l = eye_box(lmarks[0 + i * size], lmarks[1 + i * size])
+            left_eyes.append(intersection(surface, rect))
+            rect, midpoint_r = eye_box(lmarks[2 + i * size], lmarks[3 + i * size])
+            right_eyes.append(intersection(surface, rect))
+            midpoints += [midpoint_l, midpoint_r]
+        return left_eyes, right_eyes, midpoints, lmarks
+
+
+@cv.gapi.op('custom.GetStates', in_types=[cv.GArray.GMat, cv.GArray.GMat],
+            out_types=[cv.GArray.Int, cv.GArray.Int])
+class GetStates:
```

Comment:
Since `GGetStatesImpl`, it should be `GGetStates`. 

BTW, how about putting doc string for kernel & operations ?

Reply:
Changed.